### ISSUE TITLE
Layout shift when selecting between custom links and page links

### ISSUE THREAD

**Issue by richtabor** [WordPress/gutenberg#49734]
The block descriptions on these span different heights, which results in the UI shifting when selecting between the two. 

A quick fix (for the default/English) is to adjust the word count of each to be similar). 

One idea for a more substantial fix is to set a min-height value for the `BlockCard`, which helps reduce the shifting between other blocks as well. What do you think @WordPress/gutenberg-design?

### Visual

https://user-images.githubusercontent.com/1813435/231291028-4b4b4dd0-089d-4a99-a7b6-2c6b8cc6d6d1.mp4

**Comment by jasmussen** [comment]
Yep, excellent eye for detail. I recall this coming up notably with the Group block as the transforms to stack and row came up, ensuring that the transforms row sat solidly in place. In that situation we were careful to phrase the descriptions to have the same number of lines.

For the navigation block with its off-canvas editor, and indeed that same interface likely to land for the contentOnly editing interface, it would be nice to find a smarter solution here, since in those cases you'll be able to have virtually any blocks inside, not just a smaller subset. Mainly coming to mind here is maybe adding a min-height that spans, say, 2 rows of text, so that it will stay stable in _most_ cases presumably. 

I think some of the longer descriptions exist mostly on the more complicated query/page builder blocks, and in many of those cases the tabs are not present, so it's less of an issue there. 

Also want to connect some dots to a few notes I took related to the split inspector:

* Explore a semantic slot, such as an ellipsis menu next to e.g. the query loop title.
* Ellipisis menu can have a third party hook for things like a premium block “badge”,  “this is a new block” unread dot, links to meta actions (add new post, etc).
* Explore a “Styles” panel revamp that includes more context for the Styles tab, help text, variations, copy styles, etc.

It's not directly adjacent to the height issue, but it does suggest that we could zoom out and take a look at the main top area of the inspector (title, description, tabs, transforms, etc.)

**Comment by jameskoster** [comment]
> It's not directly adjacent to the height issue, but it does suggest that we could zoom out and take a look at the main top area of the inspector (title, description, tabs, transforms, etc.)

That seems like a good idea. I wonder whether the description needs to be so prominent here... generally speaking the block name alone is adequate, and some of the descriptions are a bit tautological, e.g. "Page link: A link to a page".

**Comment by jasmussen** [comment]
> I wonder whether the description needs to be so prominent here

In many cases, like the excellent tautology you mention, we can definitely do better! 

But in some cases, perhaps like the Query Loop, I think we should be as verbose as we need to in order to fully explain all the details of the block. But there might be an opportunity to separate in two categories: blocks that need only a small, perhaps 1 or at most 2 line paragraph to add character and flavor, and another of blocks that you perhaps interact with very rarely and when you do, a more elaborate description is really useful.

I find that the latter category are usually also the types of blocks that have either _mostly empty inspectors_ in which case there's no reason not to be verbose, or more complex layout blocks like the query loop that are not subjet to the split inspector behavior. In both those cases it seems fine to have long descriptions.

Here's the More block, for reference:

<img width="370" alt="Screenshot 2023-04-12 at 11 58 50" src="https://user-images.githubusercontent.com/1204802/231424548-94baff09-e245-452b-a446-5975b2482c70.png">

**Comment by jameskoster** [comment]
Yeah, there are certainly situations where the description is useful, particularly for those ambiguous blocks like 'More'. But even so, once you're familiar with a block, I don't know that you need to see the description _every time_, especially if it's several lines long. But yes, it feels like something to explore if we take a closer look at the block card.

**Comment by richtabor** [comment]
> I don't know that you need to see the description every time, especially if it's several lines long. 

An aside from this particular issue, but I was thinking it may be interesting having a toggle to show/hide block descriptions. So for those who're quite familiar with the editor can choose not to use them; similar to block breadcrumbs.

**Comment by richtabor** [comment]
> Mainly coming to mind here is maybe adding a min-height that spans, say, 2 rows of text, so that it will stay stable in most cases presumably.

That's what I'm thinking would cover _most_ of the shifting, other than blocks which have purposely detailed/long descriptions. 

> One idea for a more substantial fix is to set a min-height value for the BlockCard, which helps reduce the shifting between other blocks as well:

**Comment by jasmussen** [comment]
> An aside from this particular issue, but I was thinking it may be interesting having a toggle to show/hide block descriptions. > So for those who're quite familiar with the editor can choose not to use them; similar to block breadcrumbs.

I'm not averse to options, I like the idea that you can provide good defaults and tailor your editor. But I'd do this after exploring the other options first. Notably in the More example shown above, the inspector would be even more bare if we hid descriptions there.

**Comment by richtabor** [comment]
> One idea for a more substantial fix is to set a min-height value for the BlockCard, which helps reduce the shifting between other blocks as well. 

Let's give this a try, to reduce the shift wholesale — instead of trying to map copy length. ✅
